### PR TITLE
Jkantor/esg test data mgmt

### PR DESCRIPTION
**Pull request by jansenk**
**TL;DR -**  Add management command for setup of tests for ORA and ESG testing 

JIRA: [AU-462](https://openedx.atlassian.net/browse/AU-462)

**What changed?**

Added new management command, `create_oa_submissions_from_file`. This management command must be run from within the LMS shell after locally installing ORA.

it takes the form `./manage.py lms create_oa_submissions_from_file <course_id> <config_file_location> [--init] [--submit] [--reset]`

course id  is the target course
config file location is where submissions-config is, relative to the edx-ora folder
--init --submit and --reset are the three modes. they're mutually exclusive and one is always required.

--init goes through the config file and creates all mentioned users. usernames that submit are created and enrolled as learners. usernames mentioned in "lock" or "gradedBy" are created and enrolled as staff. --init then continues into the logic for --submit.

--submit creates a submission for every entry in the "submissions" list for each specified ORA for the given submitter. Currently, submissions are the username of the submitter, a line break, and some sentences of lorem ipsum (due to a bug in the lorem ipsum library, they look pretty janky for now). Then, if the submission has gradeData, we create a staff assessment with the given criteria-option selections and feedback.

--reset clears all submissions. Rather than clearing all users for every ORA in the course, it looks at the config file and clears all learners for all oras they appear under.

**Example config file**

```json
[
  {
    "displayName": "My ORA",
    "submissions": [
      {
        "username": "user-1",
        "lockOwner": null,
        "gradeData":{
            "gradedBy": "edx",
            "score": {
              "pointsEarned": 5,
              "pointsPossible": 10
            },
            "overallFeedback": "Lore the ipsum",
            "criteria": [
              {
                "name": "Ideas",
                "selectedOption": "Fair",
                "feedback": "some feedback for ideas"
              },
              {
                "name": "Content",
                "selectedOption": "Poor",
                "feedback": "feedback for content"
              }
            ]
        }
      },
     {
        "username": "user-2",
        "lockOwner": "edx",
        "gradeData": {}
      }
    ]
  }
]
```

**Developer Checklist**

- [ ] Reviewed the [release process](https://github.com/edx/edx-ora2/blob/master/.github/release_process.md)
- [ ] Translations and JS/SASS compiled
- [ ] Bumped version number in [setup.py](https://github.com/edx/edx-ora2/blob/a62e81a9b0d89223476967ec3c27f3557a850735/setup.py#L39) and [package.json](https://github.com/edx/edx-ora2/blob/a62e81a9b0d89223476967ec3c27f3557a850735/package.json#L3)

**Testing Instructions**

#. Create a course with one or more ORAs. 
#. Modify the example input file and save it somewhere in your edx-ora2 folder.
#. Install ORA locally by running lms and studio, then calling make install-local-ora from your ORA directory.
#. From LMS shell, run the management command with your course id, file location, and the --init flag
#. All of your users should be created and enrolled, and all specified submissions, locks, and assessments should be created.
#. From LMS shell, run the management command with your course id, file location, and the --reset flag
#. All submissions, locks, and assessments should be removed.
#. Modify values in the config file, but don't introduce any new usernames. From LMS shell, run the management command with your course id, file location, and the --submit flag
# Again, all submissions, assessment, and locks should be created.

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @edx/masters-devs-gta

### CODE REVIEW

File: openassessment/management/commands/create_oa_submissions_from_file.py [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+        )
+        parser.add_argument(
+            '--reset',
+            action='store_true',
+            help='Clear all ORA submission state from the given course'
+        )
+        parser.add_argument(
+            '--submit',
+            action='store_true',
+            help='Create submissions, assessments, and locks d'
```
Comment (nsprenkle) [inline openassessment/management/commands/create_oa_submissions_from_file.py:115]: Nit: unfinished help string

File: openassessment/management/commands/create_oa_submissions_from_file.py [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+
+    def get_display_names(self, submissions_config):
+        """
+        Returns a list of all ORA display names mentioned in the submissions config
+        """
+        display_names = []
+        for ora_config in submissions_config:
+            display_name = ora_config['displayName']
+            if display_name in display_names:
+                raise CommandError('Duplicate ORA display name found:' + display_name)
```
Comment (nsprenkle) [inline openassessment/management/commands/create_oa_submissions_from_file.py:228]: Worth mentioning this uniqueness constraint in the help/somewhere

File: openassessment/management/commands/create_oa_submissions_from_file.py [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+            self.init_ora_test_data(course_id, submissions_config)
+        elif options['submit']:
+            self.submit_ora_test_data(course_id, submissions_config)
+
+    def read_config_file(self, file_path):
+        """
+        Check that the input file exits, and attempt to open and json parse the file.
+        Returns the json parsed input file.
+        """
+        file_path = '/edx/src/edx-ora2/' + file_path
```
Comment (nsprenkle) [inline openassessment/management/commands/create_oa_submissions_from_file.py:152]: Nit: probably should use path join, and use workspace root dir if possible.

**Review by nsprenkle — APPROVED** [review]
Looks good with some minor nits, mostly for documentation

File: openassessment/management/commands/create_oa_submissions_from_file.py [hunk 4]
```diff
@@ -0,0 +1,363 @@
...
+            ignore_user_already_exists=True
+        )
+
+        print(f'Creating and enrolling {len(course_staff)} course staff')
+        call_command(
+            'create_test_users',
+            *course_staff,
+            course=course_id,
+            course_staff=True,
+            ignore_user_already_exists=True
```
Comment (jansenk) [inline openassessment/management/commands/create_oa_submissions_from_file.py:279]: This is a new flag in that command, wil link my PR as soon as it's up

Comment (jansenk) [reply openassessment/management/commands/create_oa_submissions_from_file.py:279]: https://github.com/edx/edx-platform/pull/29746

Comment (jansenk) [reply openassessment/management/commands/create_oa_submissions_from_file.py:279]: If you want to remove this line for testing, that's fine, but just know that every time you run it, all usernames mentioned in the doc will need to be globally unique.
